prompt2model/dataset_retriever/description_dataset_retriever.py

```diff
@@ -69,6 +80,15 @@ def __init__(
         self.reranking_dataset_info_file = reranking_dataset_info_file
         self.max_number_of_dataset_rows = max_number_of_dataset_rows
         self.allow_gated_datasets = allow_gated_datasets
+        self.auto_transform_data = auto_transform_data
+        self.num_points_to_transform = num_points_to_transform
+        if max_allowed_failed_transforms is None:
+            self.max_allowed_failed_transforms: int = num_points_to_transform // 3
```

Comment:
are we sure this is a reasonable default? I feel defaults based on other variables are at times problematic, similar to `max_tokens` in the generator being an issue when set as 3 x `num_prompt_tokens`

Reply:
I see your point, but I'm not sure we can set a static number here - if user wants to transform only 100 examples and we set a static number like 1000 it might be unfair. I think its strongly dependent on the num points to transform.

Reply:
@ritugala I agree with your reasoning. I think the best solution here is to add a `max_allowed_failure_rate` parameter.

Suggestion:

add
```
        max_allowed_failure_rate: float = 0.333,
)
```
to the function interface, then add:

```
        self.max_allowed_failed_transforms: int = round(num_points_to_transform * max_allowed_failure_rate)
```